.github/prompts/dotnet/async.prompt.md

```diff
@@ -0,0 +1,45 @@
+﻿This cheat sheet should serve as a quick reminder of best practices when writing asynchronous code. Following these guidelines will help you avoid common pitfalls such as unobserved exceptions, deadlocks, and unexpected UI blocking.
+
+---
+
+### ✅ DOs
+
+- **Always Await Your Tasks:**  
+  Always use the `await` keyword on async operations to ensure exceptions are captured and to avoid blocking the calling thread.  
+
+- **Use Async Task/Task<T> Methods:**  
+  Prefer returning `Task` or `Task<T>` over `async void` so that exceptions can be observed, and methods are easily composable and testable.  
+
+- **Name Methods with the "Async" Suffix:**  
+  Clearly differentiate asynchronous methods (e.g., `GetDataAsync()`) from synchronous ones.  
+
+- **Pass Cancellation Tokens:**  
+  Allow cancellation by accepting a `CancellationToken` in async methods.  
+
+- **Use ConfigureAwait(false) When Appropriate:**  
+  In library code or server-side processes, use `ConfigureAwait(false)` to avoid unnecessary context switches and potential deadlocks.  
+
+- **Keep Async Code “Async All the Way”:**  
+  Propagate async all the way from the entry point (like event handlers or controller actions) rather than mixing sync and async code.  
+
+- **Report Progress and Handle Exceptions Properly:**  
+  Use tools like `IProgress<T>` to report progress and always catch exceptions at the appropriate level when awaiting tasks.  
+
+---
+
+### ❌ DON'Ts
+
+- **Avoid async void Methods:**  
+  Except for event handlers, never use `async void` because their exceptions are not observable and they're difficult to test.  
```

Comment:
In some cases we need to override some method from platform specific and we have no option than use `async void`. For example, the `OnCreate` method on `Activity`.

```suggestion
  Except for event handlers and overriding methods, never use `async void` because their exceptions are not observable and they're difficult to test.  
```